### PR TITLE
Added ability for npcs to move diagonally 

### PR DESCRIPTION
**Pull request by BazaroZero**
Npc moves straight at player, but when runs into a wall it finds way with help of A* algo

### CODE REVIEW

File: poom/animated.py [hunk 1]
```diff
@@ -20,7 +20,9 @@ def update(self, dt: float) -> None:
     def flip_images(self) -> None:
         for index, image in enumerate(self._images):
             self._images[index] = pg.transform.flip(
-                image, flip_x=True, flip_y=False,
+                image,
```
Comment (xDiaym) [inline poom/animated.py:23]: black?

File: poom/npc.py [hunk 2]
```diff
@@ -84,22 +89,49 @@ def current_point(self) -> pg.Vector2:
         point = pg.Vector2(*self._path[self._current_point_index])
         return point + pg.Vector2(0.5)
 
+    @property
+    def _is_wall(self) -> bool:
+        owner = self._context.owner
+        direction = self._enemy.position - owner.position
+        signX = 1 if direction.x > 0 else -1
```
Comment (xDiaym) [inline poom/npc.py]: non pep8 naming

File: poom/npc.py [hunk 3]
```diff
@@ -84,22 +89,49 @@ def current_point(self) -> pg.Vector2:
...
+            not self._context._map[int(next_y)][int(owner.position.x)]
+            or not self._context._map[int(owner.position.y)][int(next_x)]
+        ):
+            return True
+        return False
+
     def update(self, dt: float) -> None:
         owner = self._context.owner
-        direction = self.current_point - owner.position
+        if self.algo == 0 and self._is_wall:
```
Comment (xDiaym) [inline poom/npc.py]: magic constant

**Review by xDiaym — CHANGES_REQUESTED** [review]

**Comment by xDiaym** [comment]
Path finding can be incapsulated in new class

File: poom/animated.py (hunk 1 above)
Comment (BazaroZero) [reply poom/animated.py:23]: Yeah

**Pull request merged**